### PR TITLE
Add version awareness to the monitoring API

### PR DESCRIPTION
**Pull request by yinan-symphony**
This commit enables the monitoring API to
  * filter a workflow's instances list by their versions
  * read a specific version of a workflow's definition

The workflow directed graph caching is also refactored within this commit.

### Description
Please put here the intent of your pull request.

### Dependencies
List the other pull requests that should be merged before/along this one.

### Checklist
- [ ] Referenced a ticket in the PR title or description
- [ ] Filled properly the description and dependencies, if any
- [ ] Unit/Integration tests updated or added
- [ ] Javadoc added or updated
- [ ] Updated the documentation in [docs folder](../docs)

### CODE REVIEW

File: workflow-bot-app/src/main/java/com/symphony/bdk/workflow/configuration/WorkflowDataSourceConfiguration.java [hunk 1]
```diff
@@ -35,4 +38,12 @@ public PlatformTransactionManager transactionManager(@Qualifier("datasource") Da
     return transactionManager;
   }
 
+  @Bean
+  @ConditionalOnProperty(value = "wdk.workflows.path", havingValue = "not-empty")
```
Comment (symphony-soufiane) [inline workflow-bot-app/src/main/java/com/symphony/bdk/workflow/configuration/WorkflowDataSourceConfiguration.java:42]: I think there is a default for wdk.workflow.path, shouldn't this be executed in this case too?

**Review by symphony-soufiane — APPROVED** [review]

**Comment by symphony-soufiane** [comment]
LGTM. The postman collection needs update. if you can update the doc to reference the postman collection would be great as we discovered there is no mention of it there.

**Comment by yinan-symphony** [comment]
> LGTM. The postman collection needs update. if you can update the doc to reference the postman collection would be great as we discovered there is no mention of it there.

added

File: workflow-bot-app/src/main/java/com/symphony/bdk/workflow/configuration/WorkflowDataSourceConfiguration.java (hunk 1 above)
Comment (yinan-symphony) [reply workflow-bot-app/src/main/java/com/symphony/bdk/workflow/configuration/WorkflowDataSourceConfiguration.java]: The default value won't impact this check.

**Pull request merged**